.github/workflows/release_notes.yml

```diff
@@ -0,0 +1,69 @@
+name: Generate Release Notes
+
+on:
+  pull_request:
+
+# on:
+#   push:
+#     branches:
+#       - main
+
+jobs:
+  send-pull-requests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.1
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Generate release notes and environment variables
+        run: |
+          ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.write_output_file
+          echo "file_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.file_name)" >> $GITHUB_ENV
+          echo "branch_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.branch_name)" >> $GITHUB_ENV
+
+      - name: Build source file
+        run: echo "source_file=/home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/${{env.file_name}}" >> $GITHUB_ENV
+
+      - name: Create branch
+        uses: dmnemec/copy_file_to_another_repo_action@main
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.RELEASE_NOTES }} # Need new token from NR open bot!
+        with:
+          source_file: "${{env.source_file}}"
+          destination_repo: 'hannahramadan/docs-website' # newrelic/docs-website
+          destination_folder: 'src/content/docs/release-notes/agent-release-notes/ruby-release-notes'
+          user_email: 'hannahr315@gmail.com' # nr-opensource-bot
+          user_name: 'hannahramadan' # opensource+bot@newrelic.com
+          destination_branch: 'develop' # dev
+          destination_branch_create: ${{env.branch_name}}
+          commit_message: 'chore(ruby agent): add release notes'
+
+      - name: Make pull request
+        uses: repo-sync/pull-request@v2
+        with:
+          github_token: ${{ secrets.RELEASE_NOTES }} # Need new token from NR open bot!
+          source_branch: ${{env.branch_name}}
+          destination_branch: "develop"
+          pr_title: "Ruby Release Notes"
+          pr_body: "This is an automated PR generated when the Ruby agent is released. Please merge as soon as possible."
+          destination_repository: "hannahramadan/docs-website" # newrelic/docs-website
+
+  delete_branch_on_fail:
+    name: Delete branch on fail
+    needs: [send-pull-requests]
+    runs-on: ubuntu-22.04
+    if: always()
+    steps:
+      - uses: actions/checkout@v3
+        with: 
+          repository: hannahramadan/docs-website # newrelic/docs-website
+          token: ${{ secrets.RELEASE_NOTES }} # Need new token from NR open bot!
+      - uses: technote-space/workflow-conclusion-action@v3
+      - run: |
+          git pull
+          git branch -a
+          git push origin --delete --force ruby_release_notes
```

Comment:
Could you walk me through what these 3 `git` commands do here?

Reply:
Ooo thanks for calling this out. I think we might only need `git push origin --delete --force ruby_release_notes` to delete the remote branch. I'll test this and update it.

Reply:
Discussed live. There is more work to be done on a job that deletes the branch upon Action failure. This work has been summarized in [#1919](https://github.com/newrelic/newrelic-ruby-agent/issues/1919)